details.js

```diff
@@ -1,204 +1,763 @@
+	this.in_style = options.in_style || new ScilabDouble();
+	this.out_style = options.out_style || new ScilabDouble();
+	this.in_label = options.in_label || new ScilabDouble();
+	this.out_label = options.out_label || new ScilabDouble();
+	this.style = options.style || new ScilabString();
+
+	return mlist(this.type, this.orig, this.sz, this.flip, this.theta, this.exprs, this.pin, this.pout, this.pein, this.peout, this.gr_i, this.id, this.in_implicit, this.out_implicit, this.in_style, this.out_style, this.in_label, this.out_label, this.style);
+
 
-function zeros(n){
-	return new Array(n+1).join('0').split('').map(parseFloat);
 }
 
 function standard_define() {
 	var sz = arguments[0];
 	var model = arguments[1];
 	var label = arguments[2];
-	var gr_i = arguments[3] || [];
-	
-	var pin = [];
-	var pout = [];
-	var pein = [];
-	var peout = [];
-	
-	var nin = model.in.length;
-	if(nin > 0){
+	var gr_i = arguments[3] || new ScilabString();
+
+	var pin = new ScilabDouble();
+	var pout = new ScilabDouble();
+	var pein = new ScilabDouble();
+	var peout = new ScilabDouble();
+
+	var nin = model[in1].length;
+	if (nin > 0) {
 		pin = zeros(nin);
 	}
-	var nout = model.out.length;
-	if(nout > 0){
+	var nout = model[out].length;
+	if (nout > 0) {
 		pout = zeros(nout);
 	}
-	var ncin = model.evtin.length;
-	if(ncin > 0){
+	var ncin = model[evtin].length;
+	if (ncin > 0) {
 		pein = zeros(ncin);
 	}
-	var ncout = model.evtout.length;
-	if(ncout > 0){
+	var ncout = model[evtout].length;
+	if (ncout > 0) {
 		peout = zeros(ncout);
 	}
-	gr_i = [gr_i, 8];
-	if(gr_i[1] == []){
-		gr_i[1] = 8;
+
+	gr_i = list(gr_i, new ScilabDouble(new data(8, 0, 0)));
+	
+	if (gr_i[1] == []) {
+		gr_i[1] = new ScilabDouble(new data(8, 0, 0));
 	}
-	if(gr_i[1] == 0){
-		gr_i[1] = [];
+	if (gr_i[1] == 0) {
+		gr_i[1] = new ScilabDouble();
 	}
+
 	var graphics_options = {
-	  sz: sz,
-	  pin: pin,
-	  pout: pout,
-	  pein: pein,
-	  peout: peout,
-	  gr_i: gr_i,
-	  exprs: label
+		sz: sz,
+		pin: pin,
+		pout: pout,
+		pein: pein,
+		peout: peout,
+		gr_i: gr_i,
+		exprs: label
 	};
 	var graphics = new scicos_graphics(graphics_options);
 	var block_options = {
-	  graphics: graphics,
-	  model: model,
-	  gui: arguments.callee.caller.name
+		graphics: graphics,
+		model: model,
+		gui: new ScilabString(new data(arguments.callee.caller.name, 0, 0))
 	};
 	return new scicos_block(block_options);
 }
 
-function scicos_link (){
-	this.xx = [];
-	this.yy = [];
-	this.id = '';
-	this.thick = [0, 0];
-	this.ct = [1, 1];
-	this.from = [];
-	this.to = [];
+function ScilabString() {
+	var i = 0;
+
+	for (i = 0; i < arguments.length; i++)
+		this['data' + i] = arguments[i];
+	if(arguments.length>1)
+	{
+		if(arguments[arguments.length-1].line>0)
+		{
+			this.height=arguments.length;
+			this.width=arguments[0].column+1;
+		}
+		else
+		{
+			this.height = arguments[0].line+1;
+			this.width = arguments.length;
+		}
+	}
+
+}
+
+function ScilabBoolean() {
```

Comment:
This should only take in boolean values and not strings - while returning however, this can return the required string